examples/webgpu_occlusion.html

```diff
@@ -0,0 +1,151 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js - WebGPU - Occlusion</title>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> - WebGPU - Occlusion<br />
+			The plane changes color if the sphere behind it is rendered
+		</div>
+
+		<script async src="https://unpkg.com/es-module-shims@1.6.3/dist/es-module-shims.js"></script>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.module.js",
+					"three/addons/": "./jsm/",
+					"three/nodes": "./jsm/nodes/Nodes.js"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+			import { nodeObject, uniform, Node, NodeUpdateType, MeshPhongNodeMaterial } from 'three/nodes';
+
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+
+			import WebGPU from 'three/addons/capabilities/WebGPU.js';
+			import WebGPURenderer from 'three/addons/renderers/webgpu/WebGPURenderer.js';
+
+			let camera, scene, renderer, controls;
+
+			class OcclusionNode extends Node {
+
+				constructor( testObject, normalColor, occludedColor ) {
+
+					super( 'vec3' );
+
+					this.updateType = NodeUpdateType.OBJECT;
+
+					this.uniformNode = uniform( new THREE.Color() );
+
+					this.testObject = testObject;
+					this.normalColor = normalColor;
+					this.occludedColor = occludedColor;
+
+				}
+
+				update( /* frame */ ) {
+
+					this.uniformNode.value.copy( this.testObject.isOccluded ? this.occludedColor : this.normalColor );
+
+				}
+
+				generate( builder, output ) {
+
+					return this.uniformNode.build( builder, output );
```

Comment:
I think, use `construct()` instead of `generate()`. I update `webgpu_instance_uniform` example too. https://github.com/mrdoob/three.js/pull/26336

```js
construct() {

	return this.uniformNode;

}
```

Reply:
You are the expert here.  I obviously copied the instance_uniform example.